lib/Parse/ParseExpr.cpp

```diff
@@ -118,7 +118,15 @@ ParserResult<Expr> Parser::parseExprAs() {
   } else if (exclaimLoc.isValid()) {
     parsed = new (Context) ForcedCheckedCastExpr(asLoc, exclaimLoc, type.get());
   } else {
-    parsed = new (Context) CoerceExpr(asLoc, type.get());
+    bool includesVarargExpansion = false;
+    auto typeRepr = type.get();
+    if (Tok.isEllipsis()) {
```

Comment:
You'll want to keep the `Tok.is(tok::oper_postfix)` check here to make sure we reject things like:

```swift
func foo(_ x: Int...) {}
foo([0] as Int ...)
```

In addition, it'll allow us to continue accepting the following:

```swift
let r = 0 as Int8...5
```

which calls the infix `...` operator.

Reply:
🤦‍♂ Good catch, thanks. Going to work on adding some more parsing tests